File: src/RoboTasker.Domain/Robots/Robot.cs
using RoboTasker.Domain.Abstractions;
using RoboTasker.Domain.Capabilities;
using RoboTasker.Domain.Robots.Enums;

namespace RoboTasker.Domain.Robots;

public class Robot : TenantEntity
{
    public string Name { get; set; } = null!;
    
    public RobotCategory Category { get; set; } = null!;
    public Guid CategoryId { get; set; }
    
    public RobotStatus Status { get; set; }
    public DateTime? LastActivity { get; set; }

    public RobotLocation Location { get; set; } = null!;
    public Guid LocationId { get; set; }
    
    public RobotCommunication Communication { get; set; } = null!;

    public IList<RobotPropertyValue> Properties { get; set; } = [];
    public IList<RobotCustomProperty> CustomProperties { get; set; } = [];
    public IList<RobotCapability> Capabilities { get; set; } = [];
}

File: src/Nerbotix.UI/src/app/models/robots/robots/robot.ts
import {RobotBase} from './robot-base';
import {RobotPropertyValue} from './robot-property-value';
import {RobotCustomProperty} from './robot-custom-property';
import {RobotCapability} from './robot-capability';
import {RobotCommunication} from './robot-communication';
import { Log } from '../../logging/log';

export class Robot extends RobotBase {
  properties!: RobotPropertyValue[];
  customProperties?: RobotCustomProperty[] | undefined;
  capabilities?: RobotCapability[] | undefined;
  logs?: Log[];
  communication!: RobotCommunication;

  constructor(obj?: Partial<Robot>) {
    super(obj);
    if (obj) {
      Object.assign(this, obj);
    }
  }
}
